Fix for Page Navigation Plugin on User Auth View

pe7er opened the pull request:
Fix for issue: https://github.com/joomla/joomla-cms/issues/5685 (see that issue for testing instruction)

The Article Pagination "Prev" & "Next" used the current article's Access Level (with "AND a.access = ' . (int) $row->access)") to navigate among articles within the same Category. That caused problems with Articles from the same Category that had different Access Levels.

This fix checks the User's authorized Access Levels (with "AND a.access IN ('.implode(",", JAccess::getAuthorisedViewLevels($user->id)).') ')" instead.

Note: using "IN" in SQL queries might be slower than "equal". I have not tested this fix on categories with huge numbers of articles!

pe7er committed:
Fix for Page Navigation Plugin on User Auth View

Fix for issue: https://github.com/joomla/joomla-cms/issues/5685

The Article Pagination "Prev" & "Next" used the current article's Access Level (with "AND a.access = ' . (int) $row->access)") to navigate among articles within the same Category. That caused problems with Articles from the same Category that had different Access Levels.

This fix checks the User's authorized Access Levels (with "AND a.access IN ('.implode(",", JAccess::getAuthorisedViewLevels($user->id)).') ')" instead.

Note: using "IN" in SQL queries might be slower than "equal". I have not tested this fix on categories with huge numbers of articles!

rdeutz commented:
You forget to make spaces before and after "." could you fix that to make travis happy, Thanks :-)

pe7er commented:
Ah, I noticed the Travis error but did not know the cause.
Thanks Robert!

pe7er committed:
Fix #2 for Page Navigation Plugin - User Auth View

Fixed code style (spaces around the concatenating dots).
Travis was not happy with my 1st PR https://github.com/joomla/joomla-cms/pull/5698

Hackwar commented:
@test successfull

Kubik-Rubik commented:
Thank you @pe7er, this PR fixes the problem.

I will merge the PR and close it!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5698">issues.joomla.org/joomla-cms/5698</a>.</sub>

pe7er commented:
Great! Thank you all for testing!
